### PR TITLE
Improve command line argument escaping

### LINKED ISSUE #1366: CodeChecker log generated json can't handle filenames with spaces
If I have a file called "lib 2.cpp" it breaks the system as CodeChecker log functionality does not escape the filename in the generated json. The generated command in json is:

```
	"command": "/usr/bin/g++-5 -Wall -c example/lib 2.cpp example/main.cpp"
```
From here, other scripts building on this json break.

### PR DESCRIPTION
**Pull request by bruntib**
In this fix we interpret escaping from the compilation database as the original
command meant. In this commit the unnecessary "smart" escaping transformations
are eliminated, because they could result in a transformation from "hello" to
\"\"\"hello\"\"\".
In CodeChecker we are storing the analyzer options as lists which can directly
be passed to Popen.

Resolves #1366
Deprecates: #1501

### CODE REVIEW

File: analyzer/codechecker_analyzer/analyzers/analyzer_clang_tidy.py [hunk 1]
```diff
@@ -117,7 +117,8 @@ def construct_analyzer_cmd(self, result_handler):
             analyzer_cmd.append("-checks='%s'" % checkers_cmdline.lstrip(','))
 
             LOG.debug(config.analyzer_extra_arguments)
-            analyzer_cmd.append(config.analyzer_extra_arguments)
+            if config.analyzer_extra_arguments:
```
Comment (gyorb) [inline analyzer/codechecker_analyzer/analyzers/analyzer_clang_tidy.py]: Is `config.analyzer_extra_arguments` a list of arguments or only a string of arguments, maybe we should change this to be a list too?

File: analyzer/codechecker_analyzer/ctu_manager.py [hunk 2]
```diff
@@ -112,7 +112,7 @@ def generate_ast(triple_arch, action, source, config, env):
 
     cmdstr = ' '.join(cmd)
     LOG.debug_analyzer("Generating AST using '%s'", cmdstr)
```
Comment (gyorb) [inline analyzer/codechecker_analyzer/ctu_manager.py]:
```suggestion
    LOG.debug_analyzer("Generating AST using '%s'", ' '.join(cmd))
```
so we would not need `cmdstr`?

**Comment by irishrover** [comment]
Thanks for this patch!
It fixes (I manually cherry-picked it and checked) irritating problem with Skia source files in Chromium.
Look at this horrible json:

`[{"directory":"/home/...","command":"../../third_party/llvm-build/Release+Asserts/bin/clang++ ...  -DSKCMS_API=__attribute__\\(\\(visibility\\(\\\"default\\\"\\)\\)\\)  -DSK_FREETYPE_MINIMUM_RUNTIME_VERSION=\\(\\(\\(FREETYPE_MAJOR\\)\\ \\*\\ 0x01000000\\)\\ \\|\\ \\(\\(FREETYPE_MINOR\\)\\ \\*\\ 0x00010000\\)\\ \\|\\ \\(\\(FREETYPE_PATCH\\)\\ \\*\\ 0x00000100\\)\\) ... -c ../../third_party/skia/src/core/SkBitmap.cpp -o obj/skia/skia_core_and_effects/SkBitmap.o","file":"../../third_party/skia/src/core/SkBitmap.cpp"}]`

`DSK_FREETYPE_MINIMUM_RUNTIME_VERSION` has spaces in its definition and they break clang-tidy without your patch.

**Comment by gyorb** [comment]
@irishrover it's great to hear that it fixes your problem

**Comment by csordasmarton** [comment]
@bruntib Please rebase your branch and resolve the merge conflict. Also check @gyorb's comments.

File: analyzer/codechecker_analyzer/ctu_manager.py (hunk 2 above)
Comment (bruntib) [reply analyzer/codechecker_analyzer/ctu_manager.py]: This variable is used a few lines lower in an error message print.

**Pull request merged**